changes/2020-07-06_clarify-streaming-encrypt-decrypt/change.md

```diff
@@ -0,0 +1,154 @@
+
+As such, this change states that once the header or a frame is serialized or parsed,
+it MUST be inputted to the signature algorithm (if included) such that it
+isn't required to remain in memory for the duration of the operation.
+
+### How and when should plaintext be released?
+
+For one-shot cases,
+where the entire message is available in memory,
+it is obvious that the operation MUST NOT release any plaintext until the
+signature (if included) is verified.
+
+For the streaming case, this gets more complicated.
+Streaming is a way to operate on large messages without requiring the entire message to be
+loaded into memory.
+This allows the processing of arbitrarily long messages with finite memory.
+However, when processing messages with a signature,
+the operation obtains plaintext in parts as it processes the message,
+but has no way of knowing whether the message is verified until the end of the operation.
+So what should we do with the unverified plaintext decrypted thus far?
+Because streaming is being used, we MUST NOT buffer the entire plaintext,
+as that would defeat the point of streaming.
+Thus, the plaintext needs to go somewhere other than memory.
+Because this is a client-side library, we MUST give it to the customer.
+
+In the case of any decryption failure,
+the operation MUST zero out any available but unreleaed plaintext.
+The customer MUST understand that all released plaintext is unverified until the operation
+completes and succeeds.
+The customer MUST discard any released plaintext if the operation fails, and MUST roll back
+any processing done with that released plaintext.
+This change states these security considerations.
+
+We should consider improvements to the format to further reduce or eliminate the need to release
+unverified plaintext in the streaming use case, however that is outside the scope of this document.
+
+### How and when does the header information become available?
+
+Any information from the header MUST NOT be released before the message header is authenticated.
+
+In the one-shot case, the operation MUST verify the whole message before outputting the header information.
+
+In the streaming case, the operation SHOULD release header information before the message is verified,
+and it MUST be interpreted as unverified data.
+
+This is important to release because a common use case for the header information is
+to check the data within it in order to ensure that the correct thing is being decrypted.
+
+Requiring customers to make such a check after already performing decryption on a large message
+could be very expensive, when they could instead immediately see possibly unexpected data within
+the header information.
+
+This change specifies the encryption context, algorithm suite ID, and parsed header as an output,
+and makes the above security considerations clear.
+
+## Reference-level Explanation
+
+The only changes needed are to the specification,
+described by the section above.
```

Comment:
I have been interpreting "Guide-level Explanation" as "for users of the ESDK" and this section as "for implementers of the ESDK".